### PR TITLE
submariner-diagnose-role: add delete permissions for pods.

### PR DESCRIPTION
**Pull request by tomer-shefler**
These are required for kubeproxy and firewall tests

### CODE REVIEW

**Comment by tomer-shefler** [comment]
related to: https://github.com/submariner-io/subctl/pull/183

File: pkg/embeddedyamls/yamls.go [hunk 1]
```diff
@@ -2783,6 +2783,7 @@ rules:
     resources:
       - pods
     verbs:
+      - delete
```
Comment (mkolesnik) [inline pkg/embeddedyamls/yamls.go:2724]: This file is auto generated, you should look at the original file `config/rbac/submariner-diagnose/role.yaml`

**Review by mkolesnik — CHANGES_REQUESTED** [review]

**Review by mkolesnik — APPROVED** [review]
LGTM, please address CI failures

**Comment by dfarrell07** [comment]
[Embedded YAMLs up-to-date](https://github.com/submariner-io/submariner-operator/runs/8074759735?check_suite_focus=true#logs) is failing. Running `make pkg/embeddedyamls/yamls.go`, [as the CI does](https://github.com/submariner-io/submariner-operator/blob/devel/.github/workflows/linting.yml#L174), should make the relevant changes for you.

File: pkg/embeddedyamls/yamls.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ spec:
   - name: v1alpha1
     schema:
       openAPIV3Schema:
-        description: Broker is the Schema for the brokers API.
+        description: Broker is the Schema for the brokers API
```
Comment (dfarrell07) [inline pkg/embeddedyamls/yamls.go]: Do you mean to remove these periods? If so, maybe that should be a different PR, or at least the reasoning documented in the commit message.

Comment (dfarrell07) [reply pkg/embeddedyamls/yamls.go]: Also this is the generated version.

File: config/rbac/submariner-diagnose/role.yaml [hunk 3]
```diff
@@ -10,6 +10,7 @@ rules:
     resources:
       - pods
     verbs:
+      - delete
```
Comment (dfarrell07) [inline config/rbac/submariner-diagnose/role.yaml:13]: Can you be more specific about what needs this permission?

File: pkg/embeddedyamls/yamls.go (hunk 2 above)
Comment (tomer-shefler) [reply pkg/embeddedyamls/yamls.go]: This dots were added by `make pkg/embeddedyamls/yamls.go`

File: config/rbac/submariner-diagnose/role.yaml (hunk 3 above)
Comment (tomer-shefler) [reply config/rbac/submariner-diagnose/role.yaml:13]: This function deletes the pods.
https://github.com/submariner-io/subctl/blob/aedd7a6e8e43e95b809f0caa96325f45213b46e9/internal/pods/schedule.go#L182
And it's being called by tests. After pods spawned by the test are done, they are deleted by the test the spawned them

**Review by tpantelis — APPROVED** [review]
